src/context.rs

```diff
@@ -31,14 +36,85 @@ impl Context {
         Context::with_values_and_filters(HashMap::new(), filters)
     }
 
-    pub fn with_values_and_filters(values: HashMap<String, Value>, filters: HashMap<String, Box<Filter>>) -> Context {
+    pub fn with_values_and_filters(values: HashMap<String, Value>,
+                                   filters: HashMap<String, Box<Filter>>) -> Context {
         Context {
-            values: values,
-            filters: filters,
+            stack: vec!(HashMap::new()),
+            globals: values,
+            filters: filters
+        }
+    }
+
+    pub fn add_filter(&mut self, name: &str, filter: Box<Filter>) {
+        self.filters.insert(name.to_owned(), filter);
+    }
+
+    pub fn get_filter<'b>(&'b self, name: &str) -> Option<&'b Box<Filter>> {
+        self.filters.get(name)
+    }
+
+    /// Creates a new variable scope chained to a parent scope.
+    pub fn push_scope(&mut self) {
```

Comment:
Done. I originally thought they might be useful outside the protected scope in with_new_scope(), but they're probably not.